include/cuco/static_multiset_ref.cuh

```diff
@@ -200,6 +200,21 @@ class static_multiset_ref
   template <typename... NewOperators>
   [[nodiscard]] __host__ __device__ auto with(NewOperators... ops) && noexcept;
 
+  /**
+   * @brief Makes a copy of the current device reference with given hash and key comparator
+   *
+   * @tparam NewKeyEqual The new key equal type
+   * @tparam NewHash The new hasher type
+   *
+   * @param key_equal New key comparator
+   * @param hash New hasher
+   *
+   * @return Copy of the current device ref
+   */
+  template <typename NewKeyEqual, typename NewHash>
+  [[nodiscard]] __host__ __device__ constexpr auto make_copy(NewKeyEqual const& key_equal,
```

Comment:
question/suggestion: `make_*` functions are always a code smell for me and require that I ask "Why can't this just be a constructor?"

Reply:
Thanks for the feedback. 

The use case is that users have a hash table ref on hand and would like to change only two data members of that ref. Using a helper function like
```cpp
auto new_ref = ref.make_copy(new_hash, new_equal);
```
can largely simplify the reconstruction process via ctor:
```cpp
auto new_ref = ref_type<Operators...>{ // how to pass variadic template parameters to the new ref
  ref.empty_key_sentinel(),
  new_equal,
  ref.probing_scheme(new_hash), //This won't work either since the new hash makes it a distinct probing scheme type
  ref.cuda_thread_scope(),
  ref.storage_ref()};
```
Another problem is with the mixin variadic template parameters and I haven't figured out a way to pass `Operators...` to the new ref without using the `make_copy` member function.

Reply:
As for naming, I don't like `make_copy` either but run out of ideas proposing a better name. Is `create` any better?

Reply:
okay, I agree that this passes the sniff test calling for something other than a constructor.